x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/ExplainDataFrameAnalyticsIT.java

```diff
@@ -81,4 +84,41 @@ public void testSourceQueryIsApplied() throws IOException {
 
         assertThat(explainResponse.getMemoryEstimation().getExpectedMemoryWithoutDisk().getKb(), lessThanOrEqualTo(1024L));
     }
+
+    public void testTrainingPercentageIsApplied() throws IOException {
+        String sourceIndex = "test-training-percentage-applied";
+        RegressionIT.indexData(sourceIndex, 100, 0);
+
+        DataFrameAnalyticsConfig config = new DataFrameAnalyticsConfig.Builder()
+            .setId("dfa-training-100-" + sourceIndex)
+            .setSource(new DataFrameAnalyticsSource(new String[] { sourceIndex },
+                QueryProvider.fromParsedQuery(QueryBuilders.matchAllQuery()),
+                null))
+            .setAnalysis(new Regression(RegressionIT.DEPENDENT_VARIABLE_FIELD,
+                BoostedTreeParams.builder().build(),
+                null,
+                100.0,
+                null))
+            .buildForExplain();
+
+        ExplainDataFrameAnalyticsAction.Response explainResponse = explainDataFrame(config);
+
+        long allDataUsedForTraining = explainResponse.getMemoryEstimation().getExpectedMemoryWithoutDisk().getBytes();
```

Comment:
Nit: You shouldn't need to use `.getBytes()` as `ByteSizeValue` is comparable so it should support the `lessThan` matcher in line 122.